### PR TITLE
Add user full name from new registration journey to SOO contact form

### PR DESCRIPTION
**Pull request by WillTaylor22**
- [x] Change has a jira ticket that has the correct status.
 - [x] Changelog entry added.

### CODE REVIEW

File: sso/models.py [hunk 1]
```diff
@@ -10,3 +10,12 @@ class SSOUser(directory_sso_api_client.models.SSOUser):
     @cached_property
     def company(self):
         return helpers.company_profile_retrieve(self.session_id)
+
+    @cached_property
+    def full_name(self):
```
Comment (richtier) [inline sso/models.py]: let's override the existing `get_full_name` method as it's a more "djangoistic" solution
```
def get_full_name():
    full_name = super().get_full_name()
    if not full_name and self.company:
        full_name = self.company['postal_full_name']
    return full_name
```
the user is not guaranteed to have a company

Comment (WillTaylor22) [reply sso/models.py]: Good point about it being djangoistic - implemented!

File: sso/models.py [hunk 2]
```diff
@@ -10,3 +10,10 @@ class SSOUser(directory_sso_api_client.models.SSOUser):
     @cached_property
     def company(self):
         return helpers.company_profile_retrieve(self.session_id)
+
+    @cached_property
```
Comment (richtier) [inline sso/models.py]: no need to use cached_property here - cached property is for when the method is performing an expensive operation such as doing an API call and we're happy to cache the result.

goold old simple function, please :)

File: sso/models.py [hunk 3]
```diff
@@ -10,3 +10,10 @@ class SSOUser(directory_sso_api_client.models.SSOUser):
...
+    def get_full_name(self):
+        full_name = super().get_full_name()
+        if not full_name and self.company and 'postal_full_name' in self.company:
```
Comment (richtier) [inline sso/models.py]: if `company` is present then `postal_full_name` will be in it. `if not full_name and self.company` is enough :)

Comment (WillTaylor22) [reply sso/models.py]: This doesn't seem very defensive?

File: sso/models.py (hunk 2 above)
Comment (WillTaylor22) [reply sso/models.py]: Done

File: sso/models.py (hunk 3 above)
Comment (WillTaylor22) [reply sso/models.py]: Anyway, removed...

**Pull request merged**